### PR TITLE
Edit to meet PyAnsys style

### PR DESCRIPTION
**Pull request by PipKat**
@plule-ansys I'm going to leave some comments and questions for you inline! Also, I couldn't run the command indicated in the README for building the doc! I received this error:

tox : The term 'tox' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ tox -e doc
+ ~~~
    + CategoryInfo          : ObjectNotFound: (tox:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

### CODE REVIEW

**Comment by PipKat** [comment]
Sorry I can't figure out the failure in the Code Style check.

File: doc/source/contributing.rst [hunk 1]
```diff
@@ -2,48 +2,50 @@
 Contributing
 ============
 
-The general guidance is available in the `Contributing` topic in the *PyAnsys
```
Comment (PipKat) [inline doc/source/contributing.rst:5]: Do you no longer need the underscore following the link in the text?

File: doc/source/contributing.rst [hunk 2]
```diff
@@ -2,48 +2,50 @@
...
-
-The test automation relies on `tox`.
+Running Tests
+-------------
+Test automation relies on `tox`.
 
-They are entirely based on mocks and do not require any external software. Run
-the tests with:
+Tests are entirely based on mocks and do not require any external software. Run
+the tests with the following code, where `py38` represents your Python version:
```
Comment (PipKat) [inline doc/source/contributing.rst:29]: Can we assume that users knows how to check to see what Python version they have? (I don't know.)

File: src/ansys/platform/instancemanagement/definition.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 @dataclass(frozen=True)
 class Definition:
-    """Definition of a product that can be started using the product instance management API.
+    """Provides a definition of a product that can be started using the PIM API.
```
Comment (PipKat) [inline src/ansys/platform/instancemanagement/definition.py:18]: Usually the docstring includes the "Parameters" section. However, you have all separate docstrings and I didn't want to make changes without asking why.

File: src/ansys/platform/instancemanagement/service.py [hunk 4]
```diff
@@ -59,18 +60,23 @@ def _build_grpc_channel(
...
+        """Build a definition from the PIM API v1 protobuf object.
+
+        Parameters
+        ----------
+        service : ServiceV1
+            Raw PIM API v1 protobuf object.       
+
+        Returns
+        -------
+        type
```
Comment (PipKat) [inline src/ansys/platform/instancemanagement/service.py:72]: Should "type" be "string"? For PyAEDT, we used "type" as a default and then replaced it with an actual data type once we determined what it was.

File: src/ansys/platform/instancemanagement/definition.py (hunk 3 above)
Comment (plule-ansys) [reply src/ansys/platform/instancemanagement/definition.py:18]: Is this section even used for class documentation?

Right now the generated doc puts the attributes like this, I think this is fine:

image

File: src/ansys/platform/instancemanagement/definition.py [hunk 5]
```diff
@@ -56,29 +55,40 @@ class Definition:
...
+        Parameters
+        ----------
+        definition : DefinitionV1
+            Raw PIM API v1 protobuf object.
+        stub : ProductInstanceManagerStub
+            Stub for the PIM instance. The default is ``None``.
+
+        Returns
+        -------
+        instance
```
Comment (plule-ansys) [inline src/ansys/platform/instancemanagement/definition.py:83]:
```suggestion
        Instance
```

File: src/ansys/platform/instancemanagement/definition.py [hunk 6]
```diff
@@ -56,29 +55,40 @@ class Definition:
...
+            PyPIM instance definition.
+
+        Raises
+        ------
+        ValueError: The raw protobuf message is not valid.
```
Comment (plule-ansys) [inline src/ansys/platform/instancemanagement/definition.py:88]:
```suggestion
        ValueError
            The raw protobuf message is not valid.
```

File: src/ansys/platform/instancemanagement/instance.py [hunk 7]
```diff
@@ -84,11 +84,15 @@ def __exit__(self, *_):
...
+        Parameters
+        ----------
+        timeout : float
+            Time in seconds to create the instance. The default is ``None``.
 
-        Returns:
-            Instance: The product instance
+        Returns
+        -------
+        instance
```
Comment (plule-ansys) [inline src/ansys/platform/instancemanagement/instance.py:94]:
```suggestion
        Instance
```

File: src/ansys/platform/instancemanagement/instance.py [hunk 8]
```diff
@@ -147,17 +156,24 @@ def build_grpc_channel(self, service_name: str = "grpc", **kwargs):
...
+        Returns
+        -------
+        grpc.Channel
+            gRPC channel preconfigured to work with the instance.
 
-        Returns:
-            grpc.Channel: gRPC channel preconfigured to work with the instance.
+        Raises
+        ------
+        ValueError: The instance does not support gRPC, or the service name is wrong.
```
Comment (plule-ansys) [inline src/ansys/platform/instancemanagement/instance.py:173]:
```suggestion
        ValueError
            The instance does not support gRPC, or the service name is wrong.
```

File: src/ansys/platform/instancemanagement/service.py [hunk 9]
```diff
@@ -44,13 +42,16 @@ def _build_grpc_channel(
     ) -> grpc.Channel:
         """Build a gRPC channel communicating with the product instance.
 
-        Args:
-            kwargs: Optional named arguments for gRPC construction.
-            They will be passed to `grpc.insecure_channel`
+        Parameteers
```
Comment (plule-ansys) [inline src/ansys/platform/instancemanagement/service.py:44]:
```suggestion
        Parameters
```

File: src/ansys/platform/instancemanagement/service.py (hunk 4 above)
Comment (plule-ansys) [reply src/ansys/platform/instancemanagement/service.py:71]:
```suggestion
        Service
            The PyPIM service
```

I think I just messed up the type, it sould have been "Service"

File: src/ansys/platform/instancemanagement/service.py [hunk 10]
```diff
@@ -59,18 +60,23 @@ def _build_grpc_channel(
...
+            Raw PIM API v1 protobuf object.       
+
+        Returns
+        -------
+        type
+            PyPIM service definition.
+        
+        Raises
+        ------
+        ValueError: The raw protobuf message is not valid.
```
Comment (plule-ansys) [inline src/ansys/platform/instancemanagement/service.py:76]:
```suggestion
        ValueError
            The raw protobuf message is not valid.
```

**Review by plule-ansys — COMMENTED** [review]
@PipKat 

Thank you! I did not realize that I was not following the standard guide, and after a quick comparison the generated documentation is **much** better.

> I couldn't run the command indicated in the README for building the doc!

Yes, it should be installed with `pip install tox`. I suggested a change adding this line.

> Sorry I can't figure out the failure in the Code Style check.

This was some pedantic complains about some while lines changes, I committed this one directly.

I've added a few recurring suggestion: In "Return" sections, using the class name with upper case letter (Client instead of client for example). This creates the correct link in the doc. Let me know if I missed something here

In "Raises" sections, format in two lines just like the other sections, which also creates the nice section with links to the error class

File: src/ansys/platform/instancemanagement/definition.py (hunk 3 above)
Comment (PipKat) [reply src/ansys/platform/instancemanagement/definition.py:18]: @plule-ansys I was trying to remember if classes had parameters before asking this--but didn't take the time to look it up. Thanks!

**Review by plule-ansys — APPROVED** [review]

**Pull request merged**